[FEATURE] map_layers variable for layout map items, tweaks to map_layers_ids

nyalldawson opened the pull request:
@etiennesky this follows up your recent map_layers_ids work, and adds a variable with a direct list of map layers (as requested by @m-kuhn ), @map_layers. I've also renamed @map_layers_ids to "@map_layer_ids", which is a more natural sounding name.

nyalldawson committed:
Show preview text for map layer values in expressions

nyalldawson committed:
[FEATURE] map_layers variable for layout map items

Just like map_layers_ids, but returns direct layer values instead
of ids

nyalldawson committed:
[needs-docs] Rename map_layers_ids to map_layer_ids

It's more natural grammar

Gustry commented:
Thanks, it was on my todo list.

I noticed that the variable "map_layers_ids" appears empty in the variable list view:
![screenshot from 2018-08-26 22-12-49](https://user-images.githubusercontent.com/1609292/44637079-3fba9a00-a97d-11e8-92af-e2ef97e3ce1b.png)

Is-it possible to display an array?

BTW, I'm @Gustry ;-)

nyalldawson committed:
Allow unformatted preview text for expression results

nyalldawson committed:
Code modernization

nyalldawson committed:
Use expression preview for variable values in variable editor widget

Means that values which are non-text, e.g. points, geometries,
lists, maps, etc show a value in the widget

nyalldawson commented:
@Gustry

> Is-it possible to display an array?

Fixed - now it uses the same expression result preview formatting as the expression widget, so all non-text variables should show nicer in the widget

nirvn commented:
Are these variables also offered in map canvas context? If not, it would be useful there too (copyright decorator, layer-dependent data-defined styling, etc.).
